kotlintest-assertions/src/main/kotlin/io/kotlintest/dsl.kt

```diff
@@ -117,29 +122,62 @@ infix fun <T> T.should(matcher: (T) -> Unit) = matcher(this)
 
 // -- specialized overrides of shouldBe --
 
-private fun equalsError(expected: Any?, actual: Any?): Throwable {
-  val expectedRepr = stringRepr(expected)
-  val actualRepr = stringRepr(actual)
-  val message = ErrorCollector.clueContext.get()+equalsErrorMessage(expectedRepr, actualRepr)
-  val throwable = junit5assertionFailedError(message, expectedRepr, actualRepr)
+internal fun equalsError(expected: Any?, actual: Any?): Throwable {
+
+  val (expectedRepr, actualRepr) = diffLargeString(stringRepr(expected), stringRepr(actual))
+  val message = ErrorCollector.clueContext.get() + equalsErrorMessage(expectedRepr, actualRepr)
+
+  val throwable = junit5AssertionFailedError(message, expectedRepr, actualRepr)
       ?: junit4comparisonFailure(expectedRepr, actualRepr)
       ?: AssertionError(message)
+
   if (Failures.shouldRemoveKotlintestElementsFromStacktrace) {
     Failures.removeKotlintestElementsFromStacktrace(throwable)
   }
   return throwable
 }
 
+/**
+ * Returns a formatted diff of the expected and actual input, unless there are no differences,
+ * or the input is too small to bother with diffing, return it returns the input as is.
+ */
+@Suppress("MoveLambdaOutsideParentheses")
+fun diffLargeString(expected: String, actual: String, minSizeForDiff: Int = 50): Pair<String, String> {
```

Comment:
I take the opinion that if a function is referentially transparent it doesn't matter if it's public.
What happens if someone wants to invoke this method? Nothing.

Reply:
Also the function is already extracted. I've just nested the functions into the proper scope (aka, a nested function).

Reply:
I think it matter if it's public. As it's not documented and not originally intended for public usage, it may bite us in the future if we decide, for example, to refactor this function to another package, or change it's implementation to some extent.

I think that if it's intended for internal use only, then it should be, even if more people could use it. 

However, if it's supposed to be a public function (I don't see why, but I don't see why not), it should be documented for users to know it exists and use it if they desire

Reply:
It can't be private or internal because I test it in another module, so it needs to be public.
But it's not meant for the general public to use.

Reply:
I'm afraid that this will force us to worry about it in the future, but I don't really mind it if you think it's ok

Reply:
In one way you're right, someone could reference this method and use it. And then if we decide to change it, they're annoyed. But it's clearly an internal method, and them calling it can't cause any issues to kotlintest.
It's a similar situation to all the com.sun classes in the jdk.

Reply:
I agree that any library will have public classes for internal use only. However, this one is in `io.kotlintest.dsl` class. That implies it's part of the DSL, doesn't it?

Is there anywhere else this function can be placed, to guarantee it's not part of any DSL that we're exposing?

Reply:
I've changed the package name to .internal so it's clear now.

Reply:
Then everything is alright to me!